tests/SemverTest.php

```diff
@@ -0,0 +1,309 @@
+//            array('1.3.0', '^1.2.3+build'),
+//            array('1.2.3', '1.2.3-pre+asdf - 2.4.3-pre+asdf'),
+//            array('1.2.3', '1.2.3pre+asdf - 2.4.3-pre+asdf'),
+//            array('1.2.3', '1.2.3-pre+asdf - 2.4.3pre+asdf'),
+//            array('1.2.3', '1.2.3pre+asdf - 2.4.3pre+asdf'),
+//            array('1.2.3-pre.2', '1.2.3-pre+asdf - 2.4.3-pre+asdf'),
+//            array('2.4.3-alpha', '1.2.3-pre+asdf - 2.4.3-pre+asdf'),
+//            array('1.2.3', '1.2.3+asdf - 2.4.3+asdf'),
+            array('1.0.0', '1.0.0'),
+//            array('0.2.4', '>=*'),
+//            array('', '1.0.0'),
+            array('1.2.3', '*'),
+            array('v1.2.3', '*'),
+            array('1.0.0', '>=1.0.0'),
+            array('1.0.1', '>=1.0.0'),
+            array('1.1.0', '>=1.0.0'),
+            array('1.0.1', '>1.0.0'),
+            array('1.1.0', '>1.0.0'),
+            array('2.0.0', '<=2.0.0'),
+            array('1.9999.9999', '<=2.0.0'),
+            array('0.2.9', '<=2.0.0'),
+            array('1.9999.9999', '<2.0.0'),
+            array('0.2.9', '<2.0.0'),
+            array('1.0.0', '>= 1.0.0'),
+            array('1.0.1', '>=  1.0.0'),
+            array('1.1.0', '>=   1.0.0'),
+            array('1.0.1', '> 1.0.0'),
+            array('1.1.0', '>  1.0.0'),
+            array('2.0.0', '<=   2.0.0'),
+            array('1.9999.9999', '<= 2.0.0'),
+            array('0.2.9', '<=  2.0.0'),
+            array('1.9999.9999', '<    2.0.0'),
+//            array('0.2.9', '<\t2.0.0'),
+            array('v0.1.97', '>=0.1.97'),
+            array('0.1.97', '>=0.1.97'),
+            array('1.2.4', '0.1.20 || 1.2.4'),
+            array('0.0.0', '>=0.2.3 || <0.0.1'),
+            array('0.2.3', '>=0.2.3 || <0.0.1'),
+            array('0.2.4', '>=0.2.3 || <0.0.1'),
+//            array('1.3.4', '||'),
+//            array('2.1.3', '2.x.x'),
+            array('1.2.3', '1.2.x'),
+            array('2.1.3', '1.2.x || 2.x'),
+            array('1.2.3', '1.2.x || 2.x'),
+            array('1.2.3', 'x'),
+//            array('2.1.3', '2.*.*'),
+            array('1.2.3', '1.2.*'),
+            array('2.1.3', '1.2.* || 2.*'),
+            array('1.2.3', '1.2.* || 2.*'),
+            array('1.2.3', '*'),
+//            array('2.1.2', '2'),
+//            array('2.3.1', '2.3'),
+            array('2.4.0', '~2.4'), // >=2.4.0 <2.5.0
+            array('2.4.5', '~2.4'),
+//            array('3.2.2', '~>3.2.1'), // >=3.2.1 <3.3.0,
+            array('1.2.3', '~1'), // >=1.0.0 <2.0.0
+//            array('1.2.3', '~>1'),
+//            array('1.2.3', '~> 1'),
+            array('1.0.2', '~1.0'), // >=1.0.0 <1.1.0,
+//            array('1.0.2', '~ 1.0'),
+//            array('1.0.12', '~ 1.0.3'),
+            array('1.0.0', '>=1'),
+            array('1.0.0', '>= 1'),
+            array('1.1.1', '<1.2'),
+            array('1.1.1', '< 1.2'),
+//            array('0.5.5', '~v0.5.4-pre'),
+//            array('0.5.4', '~v0.5.4-pre'),
```

Comment:
These two I guess fail for -pre that we don't support and the extra v? I'd just remove them.